sktime/performance_metrics/forecasting/_functions.py

```diff
@@ -135,3 +139,109 @@ def mape_loss(y_test, y_pred):
     eps = np.finfo(np.float64).eps
 
     return np.mean(np.abs(y_test - y_pred) / np.maximum(np.abs(y_test), eps))
+
+
+def evaluate(
+    forecaster, cv, y, X=None, strategy="refit", scoring=smape_loss, return_data=False
+):
+    """Evaluate forecaster using cross-validation
+
+    Parameters
+    ----------
+    forecaster : sktime.forecaster
+        Any forecaster
+    cv : sktime.SlidingWindowSplitter or sktime.ExpandingWindowSplitter
+        Splitter of how to split the data into test data and train data
+    y : pd.Series
+        Target time series to which to fit the forecaster.
+    X : pd.DataFrame, optional (default=None)
+        Exogenous variables (ignored)
+    strategy : str, optional
+        Must be "refit" or "update", by default "refit". The strategy defines
+        whether forecaster is only fitted on the first train window data and
+        then updated or always refitted.
+    scoring : function, optional
+        A score function that takes y_pred and y_test as arguments,
+        by default smape_loss
+    return_data : bool, optional
+        Returns three additional columns in the DataFrame, by default False.
+        The cells of the columns contain each a pd.Series for y_train,
+        y_pred, y_test.
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame that contains several columns with information regarding each
+        refit/update and prediction of the forecaster.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.performance_metrics.forecasting import evaluate
+    >>> from sktime.forecasting.model_selection import ExpandingWindowSplitter
+    >>> from sktime.forecasting.naive import NaiveForecaster
+    >>> y = load_airline()
+    >>> forecaster = NaiveForecaster(strategy="drift", sp=12)
+    >>> cv = ExpandingWindowSplitter(
+        initial_window=24,
+        step_length=12,
+        fh=[1,2,3,4,5,6,7,8,9,10,11,12]
+        )
+    >>> evaluate(forecaster=forecaster, y=y, cv=cv)
+    """
+    assert cv.initial_window is not None, "cv must have an initial_window"
```

Comment:
`cv` generally does not need to have an initial window, this is only required for the `update` strategy

Reply:
if I remember correctly, the initial window is also implemented as a separate method on the cv object, I now feel like this isn't a good idea any longer and something we should change perhaps

Reply:
well, I hope you followed the best practice rule "try to make interface contracts upwards compatible" then :smile:

Reply:
I added `v.initial_window` as requirement, because there seems to be some unexpected behaviour of the splitter when doing:
```python
cv = SlidingWindowSplitter(step_length=3, fh=[1,2,3])
for i, j in cv.split(y):
    # if len(i)>=cv.initial_window:
    print(i, j)
```
This ends in:
```
[] [0 1 2]
[0 1 2] [3 4 5]
[0 1 2 3 4 5] [6 7 8]
[0 1 2 3 4 5 6 7 8] [ 9 10 11]
[ 2  3  4  5  6  7  8  9 10 11] [12 13 14]
[ 5  6  7  8  9 10 11 12 13 14] [15 16 17]
[ 8  9 10 11 12 13 14 15 16 17] [18 19 20]
[11 12 13 14 15 16 17 18 19 20] [21 22 23]
[14 15 16 17 18 19 20 21 22 23] [24 25 26]
[17 18 19 20 21 22 23 24 25 26] [27 28 29]
...
```
@mloning seems the default length is 10, however why is it in the beginning <10?

Reply:
You can set `start_with_window=True` to start with the window.

Reply:
@mloning  can we make this default?

Reply:
```python
cv = SlidingWindowSplitter(initial_window=24, step_length=12, fh=[1,2,3], start_with_window=True)
for i, j in cv.split(y):
    print(i, j)
```
```
[0 1 2 3 4 5 6 7 8 9] [10 11 12]
[ 3  4  5  6  7  8  9 10 11 12] [13 14 15]
[ 6  7  8  9 10 11 12 13 14 15] [16 17 18]
[ 9 10 11 12 13 14 15 16 17 18] [19 20 21]
[12 13 14 15 16 17 18 19 20 21] [22 23 24]
...
```
Should here not the first row be like this?
`[0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23] [24 25 26]`
And the second like this?
`[35 36 37 38 39 40 41 42 43 44] [45 46 47]`

Reply:
There's a separate method for the initial window, this may be something we want to redesign